test/auth_cluster/test_raft_service_levels.py

```diff
@@ -170,3 +171,139 @@ async def test_service_levels_work_during_recovery(manager: ManagerClient):
     sls_list = await cql.run_async("LIST ALL SERVICE LEVELS")
     assert sl_v1 not in [sl.service_level for sl in sls_list]
     assert set([sl.service_level for sl in sls_list]) == set(sls + [new_sl])
+
+def default_timeout(mode):
+    if mode == "dev":
+        return "30s"
+    elif mode == "debug":
+        return "3m"
+    else:
+        "shouldn't be reached"
+
+def create_roles_stmts():
+    return [
+        "CREATE ROLE r1 WITH password='r1' AND login=true",
+        "CREATE ROLE r2 WITH password='r2' AND login=true",
+        "CREATE ROLE r3 WITH password='r3' AND login=true",
+    ]
+
+def create_service_levels_stmts():
+    return [
+        "CREATE SERVICE LEVEL sl1 WITH timeout=30m AND workload_type='interactive'",
+        "CREATE SERVICE LEVEL sl2 WITH timeout=1h AND workload_type='batch'",
+        "CREATE SERVICE LEVEL sl3 WITH timeout=30s",
+    ]
+
+def attach_service_levels_stms():
+    return [
+        "ATTACH SERVICE LEVEL sl1 TO r1",
+        "ATTACH SERVICE LEVEL sl2 TO r2",
+        "ATTACH SERVICE LEVEL sl3 TO r3",
+    ]
+
+def grant_roles_stmts():
+    return [
+        "GRANT r2 TO r1",
+        "GRANT r3 TO r2",
+    ]
+
+async def get_roles_connections(manager: ManagerClient, servers):
+    roles = ["r1", "r2", "r3"]
+    sessions = []
+
+    for role in roles:
+        cluster = manager.con_gen([s.ip_addr for s in servers], 
+            manager.port, manager.use_ssl, PlainTextAuthProvider(username=role, password=role))
+        connection = cluster.connect()
+        sessions.append((cluster, connection))
+    
+    return sessions
```

Comment:
I suggest separating clusters and sessions into two separate lists and return them. It's not clear that a list called "sessions" contains pairs of cluster and session objects.

Reply:
Perhaps returning sessions is not necessary at all if you don't use them? Maybe we can just return cluster objects.

Reply:
Sessions aren't used directly but they hold users' connections.